substratest/assets.py

```diff
@@ -152,31 +121,93 @@ def load(cls, d):
         mapper = cls.Meta.mapper
         kwargs = {}
         for k, v in d.items():
-            attr_name = mapper[k] if k in mapper else _convert(k)
-            if attr_name not in cls.__annotations__:
-                continue
-            # handle nested dataclasses;
-            # FIXME does not work for list of nested dataclasses
-            attr_type = cls.__annotations__[attr_name]
-            if dataclasses.is_dataclass(attr_type) and isinstance(v, dict):
-                v = attr_type.load(v)
-            kwargs[attr_name] = v
+            if v is not None:
+                attr_name = mapper[k] if k in mapper else _convert(k)
+                if attr_name not in cls.__annotations__:
+                    continue
+                # handle nested structures;
+                attr_type = cls.__annotations__[attr_name]
+                # because typing.List doesn't work the same way as the other types, we have to check
+                # if attr_type is a class before using issubclass()
+                if isclass(attr_type) and issubclass(attr_type, _DataclassLoader) \
+                        and isinstance(v, dict):
+                    v = attr_type.load(v)
+
+                elif hasattr(attr_type, '__origin__') and attr_type.__origin__ is list:
+                    attr_args = attr_type.__args__
```

Comment:
could be named `list_value_types` (same for next variable). This name would help to understand what this variable reprensents.